server/controllers/finance/transferts.js

```diff
@@ -0,0 +1,145 @@
+* to financial transferts and define all transferts api functions
+*
+* @required lib/db
+* @required lib/guid
+* @required journal/primarycash
+*/
+
+'use strict';
+
+var db              = require('../../lib/db'),
+    uuid            = require('../../lib/guid'),
+    journalTransfer = require('./journal/primarycash').transfer;
+
+/** get the list of all transferts */
+exports.list = getList;
+
+/** get the detail of a particular transfert */
+exports.detail = getDetail;
+
+/** create a new transfert record in the database */
+exports.create = create;
+
+/**
+* GET /transferts
+* GET /transferts?limit={number}
+*
+* @desc This function is responsible for getting transferts from the database
+*/
+function getList(req, res, next) {
+  var sql,
+      qLimit = req.query.limit;
+
+  /** Getting the transfert module id */
+  sql = 'SELECT pcm.id FROM primary_cash_module pcm WHERE pcm.text = "transfert" ';
+
+  db.exec(sql)
+  .then(function (row) {
+    /** Getting transfert records */
+    sql =
+        'SELECT pc.uuid, pc.project_id, pc.type, pc.date, pc.currency_id, pc.account_id, pc.cost, pc.description, ' +
+        'pc.cash_box_id, pc.origin_id, pc.user_id, pci.document_uuid ' +
+        'FROM primary_cash pc ' +
+        'JOIN primary_cash_item pci ON pci.primary_cash_uuid = pc.uuid ';
+
+    qLimit  = Number(qLimit);
+
+    sql += 'WHERE pc.origin_id = ? ORDER BY pc.date DESC ';
+    sql += (qLimit) ? 'LIMIT ' + Math.floor(qLimit) : '';
+
+    return db.exec(sql, [row[0].id]);
+  })
+  .then(function (rows) {
+    if (!rows.length) { next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+
+}
+
+/**
+* GET /transferts/:id
+*
+* @desc This function is responsible for getting a particular transfert record from the database
+*/
+function getDetail(req, res, next) {
+  var sql,
+      qUuid  = req.params.id;
+
+  /** Getting the transfert module id */
+  sql = 'SELECT pcm.id FROM primary_cash_module pcm WHERE pcm.text = "transfert" ';
+
+  db.exec(sql)
+  .then(function (row) {
+    /** Getting transfert records */
+    sql =
+        'SELECT pc.uuid, pc.project_id, pc.type, pc.date, pc.currency_id, pc.account_id, pc.cost, pc.description, ' +
+        'pc.cash_box_id, pc.origin_id, pc.user_id, pci.document_uuid ' +
+        'FROM primary_cash pc ' +
+        'JOIN primary_cash_item pci ON pci.primary_cash_uuid = pc.uuid ';
+
+    sql += 'WHERE pc.origin_id = ? AND pc.uuid = ? ORDER BY pc.date DESC ';
+
+    return db.exec(sql, [row[0].id, qUuid]);
+  })
+  .then(function (rows) {
+    if (!rows.length) { next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+}
+
+
+/**
+* POST /transferts
+*
+* @desc This function is responsible for creating a new transfert record in the database
+* and writing to the posting journal
+*/
+function create(req, res, next) {
+  var qData = req.body;
+
+
+  var missingRequiredParameters =
```

Comment:
Interestingly, the new database error handling should actually catch this for you, if you are running in sql_mode='STRICT_ALL_TABLES'  and have NOT NULL for all these columns (without default values).  If it is not, we should figure out why.